fendermint/eth/api/src/conv/from_eth.rs

```diff
@@ -23,37 +24,72 @@ pub fn to_fvm_message(tx: TypedTransaction) -> JsonRpcResult<Message> {
 }
 
 /// Turn a request into the DTO returned by the API.
-pub fn to_eth_transaction(
-    tx: et::Eip1559TransactionRequest,
+pub fn to_eth_transaction_response(
+    tx: &TypedTransaction,
     sig: et::Signature,
     hash: et::TxHash,
-) -> et::Transaction {
-    et::Transaction {
-        hash,
-        nonce: tx.nonce.unwrap_or_default(),
-        block_hash: None,
-        block_number: None,
-        transaction_index: None,
-        from: tx.from.unwrap_or_default(),
-        to: tx.to.and_then(|to| to.as_address().cloned()),
-        value: tx.value.unwrap_or_default(),
-        gas: tx.gas.unwrap_or_default(),
-        max_fee_per_gas: tx.max_fee_per_gas,
-        max_priority_fee_per_gas: tx.max_priority_fee_per_gas,
-        // Strictly speaking a "Type 2" transaction should not need to set this, but we do because Blockscout
-        // has a database constraint that if a transaction is included in a block this can't be null.
-        gas_price: Some(
-            tx.max_fee_per_gas.unwrap_or_default()
-                + tx.max_priority_fee_per_gas.unwrap_or_default(),
+) -> Result<et::Transaction, JsonRpcError> {
+    match &tx {
```

Comment:
This all looks very repetitive. Can't you populate the common values from `TransactionEssentials` via [`essentials()`](https://github.com/foundry-rs/foundry/blob/eef87de35375311f134fb4261c18455ced8022a5/crates/anvil/core/src/eth/transaction/mod.rs#L690)? And then add the specifics?

Isn't there a method to normalize typed transactions into the Transaction response type in ethers?

Reply:
> Isn't there a method to normalize typed transactions into the Transaction response type in ethers?
From what I found, not really. My guess is the Transaction response contains transaction index, which does not make sense to do the conversion using `From`.